include/boost/geometry/algorithms/detail/envelope/range_of_boxes.hpp

```diff
@@ -258,29 +249,27 @@ struct envelope_range_of_boxes
 
         BOOST_GEOMETRY_ASSERT(! boost::empty(range_of_boxes));
 
-        iterator_type it_min = std::min_element(boost::begin(range_of_boxes),
-                                                boost::end(range_of_boxes),
-                                                latitude_less<min_corner>());
-        iterator_type it_max = std::max_element(boost::begin(range_of_boxes),
-                                                boost::end(range_of_boxes),
-                                                latitude_less<max_corner>());
+        auto it_min = std::min_element(boost::begin(range_of_boxes),
+                                       boost::end(range_of_boxes),
+                                       latitude_less<min_corner>());
+        auto it_max = std::max_element(boost::begin(range_of_boxes),
+                                       boost::end(range_of_boxes),
+                                       latitude_less<max_corner>());
 
         coordinate_type const min_longitude = constants::min_longitude();
         coordinate_type const max_longitude = constants::max_longitude();
         coordinate_type const period = constants::period();
 
         interval_range_type intervals;
-        for (iterator_type it = boost::begin(range_of_boxes);
-             it != boost::end(range_of_boxes);
-             ++it)
+        for (auto const& box : range_of_boxes)
```

Comment:
While it's true that currently this function is used only internally I wouldn't be so eager to replace this loop with range-based version. It's because technically we could introduce MultiBox concept in the future. In general I propose to leave normal loops in algorithms taking ranges as template parameters as if they were geometries because replacing them may have unintended consequences for users.

Reply:
:heavy_check_mark: good point, reverted (keeping `box`)